spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/rxjava2/SleuthRxJava2SchedulersHandler.java

```diff
@@ -0,0 +1,145 @@
+package org.springframework.cloud.sleuth.instrument.rxjava2;
+
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.cloud.sleuth.TraceKeys;
+
+import brave.Span;
+import brave.Tracer;
+import io.reactivex.functions.Function;
+import io.reactivex.plugins.RxJavaPlugins;
+import io.reactivex.schedulers.SchedulerRunnableIntrospection;
+
+/**
+ * {@link RxJavaPlugins} setup schedule handler into tracing for all schedulers
+ * representation.
+ *
+ * @author Łukasz Guz
+ * @author Jakub Pyda
+ */
+class SleuthRxJava2SchedulersHandler {
+
+	private static final Log log = LogFactory
+			.getLog(SleuthRxJava2SchedulersHandler.class);
+
+	private static final String RXJAVA_COMPONENT = "rxjava2";
+
+	SleuthRxJava2SchedulersHandler(Tracer tracer, TraceKeys traceKeys,
+			List<String> threadsToSample) {
+		try {
+			Function<? super Runnable, ? extends Runnable> delegate = RxJavaPlugins
+					.getScheduleHandler();
+			if (delegate instanceof SleuthRxJava2SchedulersHandler.ScheduleHandler) {
+				return;
+			}
+			logRegisterRxJavaPlugins();
+			RxJavaPlugins.reset();
```

Comment:
just reviewing this closer while pulling out current trace context code.

ps this resets all plugins to null/default.. is this what you wanted to do? What if other plugins besides scheduler were present?

Reply:
I little based on plugin to RxJava1 therefore I left this line.